python/beeai_framework/adapters/beeai_platform/agents/agent.py

```diff
@@ -86,9 +96,18 @@ async def error_event(data: A2AAgentErrorEvent, event: EventMeta) -> None:
             )
 
         message = self._agent.convert_to_a2a_message(input)
-        message.metadata = (message.metadata or {}) | await self._get_metadata()
+        message.metadata = (message.metadata or {}) | (
+            beeai_platform_context.metadata
+            if (beeai_platform_context and beeai_platform_context.metadata)
+            else await self._get_metadata()
+        )
+        message.context_id = (
+            self._platform_context.id
+            if self._platform_context
+            else (beeai_platform_context.context.context_id if beeai_platform_context else None)
+        )
```

Comment:
We could extend `convert_to_a2a_message` to take `metadata` and `context_id` as overrides